x-pack/plugins/stack_alerts/server/alert_types/es_query/alert_type.ts

```diff
@@ -159,6 +161,19 @@ export function getAlertType(
         { name: 'index', description: actionVariableContextIndexLabel },
       ],
     },
+    useSavedObjectReferences: {
+      extractReferences: (params) => {
```

Comment:
I think this should only be applied to the new ES query rule type, not the legacy. That should be differentiated by `params.searchType` right? However, when I tried to create a legacy ES query rule type and inspected the params being passed into this function, I see `params.searchType` is `null` and it is also saved as `null` in the saved object.

Reply:
@Dmitriynj in our migration 

https://github.com/elastic/kibana/blob/2c091706c051edc8d4b83c41576a62d76ef8468c/x-pack/plugins/alerting/server/saved_objects/migrations.ts#L722

so I think we should no omit the searchType for the "legacy" rule, but use that type here

Reply:
reviewed the changes, LGTM